vavr/src/main/java/io/vavr/control/Option.java

```diff
@@ -361,8 +371,51 @@ default T getOrElse(Supplier<? extends T> supplier) {
     }
 
     /**
-     * Maps the value and wraps it in a new {@code Some} if this is a {@code Some}, returns {@code None}.
+     * Maps the value to a new {@code Option} if this is a {@code Some}, otherwise returns {@code None}.
+     * In the case the given {@code mapper} throws, {@code None} will be returned.
+     * <p>
+     * Examples:
+     * <pre>{@code
+     * // = None
+     * Option.<Integer> none().flatMapTry(i -> 1 / i);
+     *
+     * // = Some(0)
+     * Option.some(2).flatMapTry(i -> Option.of(1 / i));
      *
+     * // = None
+     * Option.some(0).flatMapTry(i -> Option.of(1 / i)); // division by zero
```

Comment:
(I'm into  definitions and...) I'm not sure this is a fantastic example (for my use case anyhow) - but quite possibly is for this change...

`i -> Option.of(1/i)` is really a `PartialFunction<Int, Int>`. I'd be writing this as `Option.some(0).flatMap(reciprocalFunction.lift())`. I'm not sure there's a really valid semantic use case for this function - certainly it's a shorter path, but I'm not sure that's always a good thing.

`flatMapTry(function)` is the same as `flatMap(t -> Try.of(() -> function.apply(t)).toOption())` right? Or rather `flatMap(t --> function.liftTry().apply(t).toOption())`, or `map(liftTry(function)).flatMap(Try::toOption)`

Seems like this change is helpful, but I'd want to see the equivalent expansion in the docs.

Reply:
Thank you! Yoy are right, it is a partial function.
The discussion in the original issue shows that I've missed the point here. What we really need is a mapOrThrow() that takes a CheckedFunction1 and re-throws any exception.
I will update the PR accordingly.